config/v1/types_ingress.go

```diff
@@ -91,7 +91,7 @@ type IngressSpec struct {
 type ConsumingUser string
 
 // Hostname is an alias for hostname string validation.
-// +kubebuilder:validation:Format=hostname
+// +kubebuilder:validation:Pattern=`^(?=.{1,256}$)(([a-z0-9][-a-z0-9]{0,61}[a-z0-9]|[a-z0-9]{1,63})[\.]){0,}([a-z0-9][-a-z0-9]{0,61}[a-z0-9]|[a-z0-9]{1,63})$`
```

Comment:
The max should be 253 characters:
```suggestion
// +kubebuilder:validation:Pattern=`^(?=.{1,253}$)(([a-z0-9][-a-z0-9]{0,61}[a-z0-9]|[a-z0-9]{1,63})[\.]){0,}([a-z0-9][-a-z0-9]{0,61}[a-z0-9]|[a-z0-9]{1,63})$`
```
Alternatively, this should do the same:
```suggestion
// +kubebuilder:validation:Pattern=`^(([a-z0-9][-a-z0-9]{0,61}[a-z0-9]|[a-z0-9]{1,63})[\.]){0,}([a-z0-9][-a-z0-9]{0,61}[a-z0-9]|[a-z0-9]{1,63})$`
// +kubebuilder:validation:MaxLength=253
```